### DISCUSSION TITLE
Diff of prompt versions

### DISCUSSION THREAD

**Discussion by dylanjcastillo** [langfuse/langfuse#3131] [Ideas]
### Describe the feature or potential improvement

This feature request proposes adding a prompt comparison functionality, allowing users to see the differences between various prompt versions using a "diff"-like approach. This would be especially valuable when iterating on prompts and fine-tuning their structure for improved results, making it easier to track changes and understand the impact of modifications.

**Comment by marcklingen** [comment]
Thanks for sharing this!

**Comment by clemra** [comment]
+ +1 from : https://x.com/hungquangphan/status/1872238497995837630

**Comment by chrispe** [comment]
I've submitted a PR (https://github.com/langfuse/langfuse/pull/4846) that enables the user to compare the current selected prompt to it's previous version.

@marcklingen Feel free to have a look and let me know what you think. I would love to get this feature working on Langfuse.

![Screenshot 2024-12-30 at 10 22 05 AM](https://github.com/user-attachments/assets/088f80db-d87f-4cd8-a16b-a8e5472f7943)

**Reply by marcklingen** [reply]
thanks for submitting the PR, I'll have a look!

**Comment by kajsatretow** [comment]
+1

**Comment by marcklingen** [comment]
@hassiebp released this today based on the initial implementation of @chrispe

Diffs are now available when creating new versions and when browsing existing versions.

Example on existing prompt versions:

![405724579-9dec41e7-a1ac-4ef7-afbd-321f299449fb](https://github.com/user-attachments/assets/7edc6e57-7600-475a-ad1f-3e5552be341e)

**Reply by chrispe** [reply]
Looks great, thank you @hassiebp!
